### PR TITLE
Restructured the App flow

### PR DESCRIPTION
**Pull request by tnovak-olin**

### CODE REVIEW

File: Clew/Assets.xcassets/Align.imageset/Contents.json [hunk 1]
```diff
@@ -2,17 +2,17 @@
...
     {
       "idiom" : "universal",
-      "filename" : "Align.png",
+      "filename" : "landmark2.png",
       "scale" : "1x"
     },
     {
       "idiom" : "universal",
-      "filename" : "align@2x.png",
+      "filename" : "landmark2@2x.png",
```
Comment (Inkering) [inline Clew/Assets.xcassets/Align.imageset/Contents.json]: Is this is just renaming for consistency?

File: Clew/Controllers/StateViews/PauseTrackingController.swift [hunk 2]
```diff
@@ -97,20 +131,20 @@ class PauseTrackingController: UIViewController, UIScrollViewDelegate {
         label.lineBreakMode = NSLineBreakMode.byWordWrapping
 
         /// initialize buttons with some basic size constaints
-        enterLandmarkDescriptionButton = UIButton.makeConstraintButton(view,
+        enterAnchorPointDescriptionButton = UIButton.makeConstraintButton(view,
                                                                   alignment: UIConstants.ButtonContainerHorizontalAlignment.left,
                                                                   appearance: UIConstants.ButtonAppearance.imageButton(image: UIImage(named: "Describe")!),
-                                                                  label: "Enter text to help you remember this landmark")
+                                                                  label: NSLocalizedString("enterAnchorPointDescriptionButtonAccessabilityLabel", comment: "This is the accessability label for the button which allows the user to save a text based description of their anchor point when saving a route. This feature should allow the user to more easily realign with their anchorpoint."))
```
Comment (paulruvolo) [inline Clew/Controllers/StateViews/PauseTrackingController.swift]: accessibility is misspelled

File: Clew/Controllers/StateViews/PauseTrackingController.swift [hunk 3]
```diff
@@ -97,20 +131,20 @@ class PauseTrackingController: UIViewController, UIScrollViewDelegate {
...
         
         recordVoiceNoteButton = UIButton.makeConstraintButton(view,
                                                          alignment: UIConstants.ButtonContainerHorizontalAlignment.right,
                                                          appearance: UIConstants.ButtonAppearance.imageButton(image: UIImage(named: "VoiceNote")!),
-                                                         label: "Record audio to help you remember this landmark")
+                                                         label: NSLocalizedString("enterAnchorPointVoiceNoteButtonAccessabilityLabel", comment: "This is the accessability label for the button which allows the user to save a voice note description of their anchor point when saving a route. This feature should allow the user to more easily realign with their anchorpoint."))
```
Comment (paulruvolo) [inline Clew/Controllers/StateViews/PauseTrackingController.swift]: accessibility is misspelled

File: Clew/Controllers/StateViews/PauseTrackingController.swift [hunk 4]
```diff
@@ -97,20 +131,20 @@ class PauseTrackingController: UIViewController, UIScrollViewDelegate {
...
         
         confirmAlignmentButton = UIButton.makeConstraintButton(view,
                                                           alignment: UIConstants.ButtonContainerHorizontalAlignment.center,
                                                           appearance: UIConstants.ButtonAppearance.imageButton(image: UIImage(named: "Align")!),
-                                                          label: "Start alignment countdown")
+                                                          label: NSLocalizedString("startAlignmentCountdownButtonAccessabilityLabel", comment: "this is athe accessability label for the button which allows the user to start an alignment procedure when saving an anchor point"))
```
Comment (paulruvolo) [inline Clew/Controllers/StateViews/PauseTrackingController.swift]: accessibility is misspelled

File: Clew/Controllers/StateViews/RecordPathController.swift [hunk 5]
```diff
@@ -76,29 +131,35 @@ class RecordPathController: UIViewController {
...
-        stackView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: UIConstants.yButtonFrameMargin).isActive = true
-        stackView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -UIConstants.yButtonFrameMargin).isActive = true
-        stackView.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -12).isActive = true
+        stackView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 5).isActive = true
+        stackView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -5).isActive = true
+        stackView.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: 20).isActive = true
+        stackView.topAnchor.constraint(equalTo: view.topAnchor, constant: 115).isActive = true
         
         if let parent: UIViewController = parent {
+
```
Comment (paulruvolo) [inline Clew/Controllers/StateViews/RecordPathController.swift]: Verify that this TODO still applies

File: Clew/Controllers/StateViews/NameSavedRouteController.swift [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    /// button for finalizing your saved route name
+    var saveRouteButton: UIButton!
+    
+    /// Label for description of what to do to name your route
+    var label: UILabel!
+    
+    /// Text Field for typing the name you want to save your route as.
+    var textField: UITextField!
+    
+    /// The ARWorldMap represented as an Any? to allow for backwards compatibility with iOS 11.3 and 11.4don
```
Comment (Inkering) [inline Clew/Controllers/StateViews/NameSavedRouteController.swift]: 11.4don?

File: Clew/Controllers/StateViews/NameSavedRouteController.swift [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+        label.textColor = UIColor.white
+        label.textAlignment = .center
+        label.numberOfLines = 0
+        label.lineBreakMode = NSLineBreakMode.byWordWrapping
+        label.lineBreakMode = .byWordWrapping
+        label.font = UIFont.preferredFont(forTextStyle: .title1)
+        label.text = mainText
+        label.tag = UIView.mainTextTag
+        label.widthAnchor.constraint(equalToConstant: UIScreen.main.bounds.size.width / 1.1).isActive = true
+//        label.heightAnchor.constraint(equalToConstant: UIScreen.main.bounds.size.width / 1.9).isActive = true
```
Comment (Inkering) [inline Clew/Controllers/StateViews/NameSavedRouteController.swift]: delete this comment

File: Clew/Controllers/StateViews/RecordPathController.swift.orig [hunk 8]
```diff
@@ -0,0 +1,150 @@
+//
```
Comment (paulruvolo) [inline Clew/Controllers/StateViews/RecordPathController.swift.orig]: Should we just get rid of this file?

File: Clew/Controllers/StateViews/ResumeTrackingConfirmController.swift [hunk 9]
```diff
@@ -91,41 +91,41 @@ class ResumeTrackingConfirmController: UIViewController, UIScrollViewDelegate {
...
-                                                       label: "Play recorded voice note")
+                                                       label: NSLocalizedString("playVoiceNoteButtonAccessabilityLabel", comment: "This is the accessability label for the button which allows the user to replay their recorded voice note when loading an anchor point."))
         
         // MARK: ConfirmAlignmentButton
         confirmAlignmentButton = UIButton.makeConstraintButton(view,
                                                                alignment: UIConstants.ButtonContainerHorizontalAlignment.center,
                                                                appearance: UIConstants.ButtonAppearance.imageButton(image: UIImage(named: "Align")!),
-                                                               label: "Start alignment countdown")
+                                                               label: NSLocalizedString("startAlignmentCountdownButtonAccessabilityLabel", comment: "this is athe accessability label for the button which allows the user to start an alignment procedure when saving an anchor point"))
```
Comment (paulruvolo) [inline Clew/Controllers/StateViews/ResumeTrackingConfirmController.swift]: accessibility is misspelled

File: Clew/Controllers/StateViews/ResumeTrackingController.swift [hunk 10]
```diff
@@ -88,7 +88,7 @@ class ResumeTrackingController: UIViewController {
         resumeButton = UIButton.makeConstraintButton(view,
                                                 alignment: UIConstants.ButtonContainerHorizontalAlignment.center,
                                                 appearance: UIConstants.ButtonAppearance.imageButton(image: UIImage(named: "Resume")!),
-                                                label: NSLocalizedString("Resume", comment: "Resume paused route"))
+                                                label: NSLocalizedString("resumePausedRouteBUttonAccessabilityLabel", comment: "Accessability label for button allowing the user to resume a paused route"))
```
Comment (paulruvolo) [inline Clew/Controllers/StateViews/ResumeTrackingController.swift]: accessibility is misspelled

File: Clew/Controllers/StateViews/RouteRatingController.swift [hunk 11]
```diff
@@ -96,11 +91,11 @@ class RouteRatingController: UIViewController {
         thumbsDownButton = UIButton.makeConstraintButton(view,
                                                     alignment: UIConstants.ButtonContainerHorizontalAlignment.leftcenter,
                                                     appearance: UIConstants.ButtonAppearance.imageButton(image: UIImage(named: "thumbs_down")!),
-                                                    label: NSLocalizedString("Bad", comment: "Unsatisfactory service"))
+                                                    label: NSLocalizedString("badServiceButtonAccessabilityLabel", comment: "The accessability label on the button denoting that the route navigation provided unsatisfactory service"))
```
Comment (paulruvolo) [inline Clew/Controllers/StateViews/RouteRatingController.swift]: accessibility is misspelled

File: Clew/Controllers/StateViews/RouteRatingController.swift [hunk 12]
```diff
@@ -96,11 +91,11 @@ class RouteRatingController: UIViewController {
...
         thumbsUpButton = UIButton.makeConstraintButton(view,
                                                   alignment: UIConstants.ButtonContainerHorizontalAlignment.rightcenter,
                                                   appearance: UIConstants.ButtonAppearance.imageButton(image: UIImage(named: "thumbs_up")!),
-                                                  label: NSLocalizedString("Good", comment: "Satisfactory service"))
+                                                  label: NSLocalizedString("goodServiceButtonAccessabilityLabel", comment: "The accessability label on the button denoting that the route navigation provided satisfactory service"))
```
Comment (paulruvolo) [inline Clew/Controllers/StateViews/RouteRatingController.swift]: accessibility is misspelled

File: Clew/Controllers/StateViews/RecordPathController.swift.orig (hunk 8 above)
Comment (Inkering) [inline Clew/Controllers/StateViews/RecordPathController.swift.orig]: just delete this .orig file

File: Clew/Controllers/StateViews/StartNavigationController.swift [hunk 13]
```diff
@@ -21,44 +21,107 @@ class StartNavigationController: UIViewController {
...
+        
+        /// configure label: Zero lines + Word Wrapping
+        label.numberOfLines = 0
+        label.lineBreakMode = NSLineBreakMode.byWordWrapping
         startNavigationButton = UIButton.makeConstraintButton(view,
-                                                         alignment: UIConstants.ButtonContainerHorizontalAlignment.center,
+                                                         alignment: UIConstants.ButtonContainerHorizontalAlignment.left,
                                                          appearance: UIConstants.ButtonAppearance.imageButton(image: UIImage(named: "StartNavigation")!),
-                                                         label: NSLocalizedString("Start navigation", comment: "The name of the button that allows user to start navigating."))
+                                                         label: NSLocalizedString("startReturnNavigationButtonAccessabilityLabel", comment: "The accessability label for the button that allows user to start navigating back along their route."))
```
Comment (paulruvolo) [inline Clew/Controllers/StateViews/StartNavigationController.swift]: accessibility is misspelled

File: Podfile.lock.orig [hunk 14]
```diff
@@ -0,0 +1,140 @@
+PODS:
```
Comment (paulruvolo) [inline Podfile.lock.orig]: this is a merge conflict.  It should be deleted.

File: Clew/Controllers/ViewController.swift [hunk 15]
```diff
@@ -1343,41 +1324,81 @@ class ViewController: UIViewController, ARSCNViewDelegate, SRCountdownTimerDeleg
...
+        print("starting pause procedure", creatingRouteAnchorPoint)
+        state = .startingPauseProcedure
     }
     
     /// handles the user pressing the stop recording button.
     ///
     /// - Parameter sender: the button that generated the event
     @objc func stopRecording(_ sender: UIButton) {
-        if beginRouteLandmark.transform != nil {
+        /*
```
Comment (Inkering) [inline Clew/Controllers/ViewController.swift]: what's going on here with this comment?

File: Clew/Controllers/ViewController.swift [hunk 16]
```diff
@@ -1343,41 +1324,81 @@ class ViewController: UIViewController, ARSCNViewDelegate, SRCountdownTimerDeleg
...
-                getRouteNameAndSaveRouteHelper(mapAsAny: nil)
+                //DELETEPOINT
+                //getRouteNameAndSaveRouteHelper(mapAsAny: nil)
             }
-        }
+        }*/
 
         isResumedRoute = false
-        state = .readyToNavigateOrPause(allowPause: true)
+        //TODO add conditional for one way route
```
Comment (Inkering) [inline Clew/Controllers/ViewController.swift]: Is this todo still relevant?

File: Clew/Controllers/ViewController.swift [hunk 17]
```diff
@@ -2072,28 +2132,29 @@ class ViewController: UIViewController, ARSCNViewDelegate, SRCountdownTimerDeleg
...
-            // resetting the origin is needed in the case when we realigned to a saved route
-            session.setWorldOrigin(relativeTransform: simd_float4x4.makeTranslation(0,0,0))
             if case .readyForFinalResumeAlignment = state {
                 // this will cancel any realignment if it hasn't happened yet and go straight to route navigation mode
                 rootContainerView.countdownTimer.isHidden = true
                 isResumedRoute = true
                 
+                isAutomaticAlignment = true
+                
+                ///PATHPOINT: Auto Alignment -> resume route
```
Comment (Inkering) [inline Clew/Controllers/ViewController.swift]: what is a PATHPOINT?

File: Clew/Controllers/ViewController.swift.orig [hunk 18]
```diff
@@ -0,0 +1,2227 @@
+//
```
Comment (Inkering) [inline Clew/Controllers/ViewController.swift.orig]: delete this .orig file

File: Clew/Views/RootContainerView.swift [hunk 19]
```diff
@@ -103,9 +104,10 @@ class RootContainerView: UIView {
         helpButton.isAccessibilityElement = true
         helpButton.setTitle("Help", for: .normal)
         helpButton.titleLabel?.font = UIFont.systemFont(ofSize: 24.0)
-        helpButton.accessibilityLabel = "Help"
+        helpButton.accessibilityLabel = NSLocalizedString("helpButtonAccessabilityLabel", comment: "This is the accessability label for the help button")
```
Comment (Inkering) [inline Clew/Views/RootContainerView.swift]: do we want to get rid of this comment??

File: Podfile.lock.orig (hunk 14 above)
Comment (Inkering) [inline Podfile.lock.orig]: delete this .orig file

File: Clew/Assets.xcassets/Align.imageset/Contents.json (hunk 1 above)
Comment (tnovak-olin) [reply Clew/Assets.xcassets/Align.imageset/Contents.json]: this actually changes the file form the carpenter's square image to a thumbtack image

File: Clew/Controllers/ViewController.swift (hunk 17 above)
Comment (tnovak-olin) [reply Clew/Controllers/ViewController.swift]: this is the location where the app changes state to the next section of flow. This gives users a way to easily find the locations of a change of state of the app.

**Pull request merged**